### ISSUE TITLE
Only move data files to an external disk

### ISSUE THREAD

**Issue by sim-san** [nextcloud/all-in-one#271]
In the Readme, you mentioned the possibility to move the complete docker directory on an external disk. [how-to-store-the-filesinstallation-on-a-separate-drive](https://github.com/nextcloud/all-in-one#how-to-store-the-filesinstallation-on-a-separate-drive).

_Do you plan to support an option to only store the data files on an external mount like the nextcloud docker image ?_

```
$ docker run -d \
-v nextcloud:/var/www/html \
-v apps:/var/www/html/custom_apps \
-v config:/var/www/html/config \
-v data:/var/www/html/data \
-v theme:/var/www/html/themes/<YOUR_CUSTOM_THEME> \
nextcloud
```
see [nextcloud docker | persistent data ](https://github.com/nextcloud/docker#persistent-data)

**Background**:
I want to host Nextcloud AIO on an Azure VM and use an blob storage mounted with [blobfuse](https://github.com/Azure/azure-storage-fuse) to have an cheap storage location. In order not to endanger the stability of the containers, I would like to mount only the data files on the blobfuse mount

**Comment by szaimen** [comment]
Hello,

that sounds interesting what you are trying to achieve here.
Such an option is currently not planned.

What about using this new option: https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host to mount a custom directory into the container and make it in Nextcloud available using local external storage? Would that work?

Apart from that we are actually already using docker volumes as described in the linked readme about persistent storage. What you are probably looking for is the docker volume nextcloud_aio_nextcloud_data which is by default located here: /var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/

**Comment by sim-san** [comment]
>  If I understand your request correctly, you want to change the default datadir of Nextcloud to a custom directory on your host. Correct?
Yes, correct.

> What you are probably looking for is the docker volume nextcloud_aio_nextcloud_data which is by default located here: /var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/
yes, this would what I am looking for.

**Comment by sim-san** [comment]
> What about using this new option: https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host to mount a custom directory into the container and make it in Nextcloud available using local external storage? Would that work? (it will like this though not be covered by the built-in backup solution).

Is it possible that all data of all users are on the external storage and it does not look to the users that this is an extra folder ?

**Comment by szaimen** [comment]
> Is it possible that all data of all users are on the external storage and it does not look to the users that this is an extra folder ?

Yes but it is not as straight forward as using the normal datadir and not covered by the built in backup solution as I said.

**Comment by david-steg** [comment]
I would be also very interessted about having the possibility to change the docker Volume "nextcloud_aio_nextcloud_data" to a Mount on the Host. Like /mnt/nextcloud/data instead of  /var/lib/docker/volumes/nextcloud_aio_nextcloud_data/_data/.
In this case I could properly mount a NFS or CIFS Share to this folder.
Mounting a Share into the Docker Volume Path doesn't sound like a clean solution.
Would be that possible with a ENV. VAR. like "NEXTCLOUD_DATA_DIR"?

**Comment by szaimen** [comment]
> Would be that possible with a ENV. VAR.

Yes, it will be implemented with an environmental variable 👍

**Comment by szaimen** [comment]
This is now implemented with v0.7.0. Here are some docs on this: https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
